src/script/script.h

```diff
@@ -199,41 +199,27 @@ class CScriptNum
  * throwing an exception if arithmetic is done or the result is interpreted as an integer.
  */
 public:
+    static const size_t MAXIMUM_ELEMENT_SIZE = 4;
 
-    explicit CScriptNum(const int64_t& n)
-    {
-        m_value = n;
-    }
+    explicit CScriptNum(const int64_t &n) { m_value = n; }
 
-    static const size_t nDefaultMaxNumSize = 4;
-
-    explicit CScriptNum(const std::vector<unsigned char>& vch, bool fRequireMinimal,
-                        const size_t nMaxNumSize = nDefaultMaxNumSize)
-    {
+    explicit CScriptNum(const std::vector<uint8_t> &vch, bool fRequireMinimal,
+                        const size_t nMaxNumSize = MAXIMUM_ELEMENT_SIZE) {
         if (vch.size() > nMaxNumSize) {
             throw scriptnum_error("script number overflow");
         }
-        if (fRequireMinimal && vch.size() > 0) {
-            // Check that the number is encoded with the minimum possible
-            // number of bytes.
-            //
-            // If the most-significant-byte - excluding the sign bit - is zero
-            // then we're not minimal. Note how this test also rejects the
-            // negative-zero encoding, 0x80.
-            if ((vch.back() & 0x7f) == 0) {
-                // One exception: if there's more than one byte and the most
-                // significant bit of the second-most-significant-byte is set
-                // it would conflict with the sign bit. An example of this case
-                // is +-255, which encode to 0xff00 and 0xff80 respectively.
-                // (big-endian).
-                if (vch.size() <= 1 || (vch[vch.size() - 2] & 0x80) == 0) {
-                    throw scriptnum_error("non-minimally encoded script number");
-                }
-            }
+        if (fRequireMinimal && !IsMinimallyEncoded(vch, nMaxNumSize)) {
+            throw scriptnum_error("non-minimally encoded script number");
         }
         m_value = set_vch(vch);
     }
 
+    static bool IsMinimallyEncoded(
+        const std::vector<uint8_t> &vch,
+        const size_t nMaxNumSize = CScriptNum::MAXIMUM_ELEMENT_SIZE);
+
```

Comment:
A comment like this for MinimallyEncode would help a lot
`// Returns true when the input encoding was changed`